doc/reference/reference_lua/http.rst

```diff
@@ -1,353 +1,1141 @@
-    It affects libcurl `CURLMOPT_MAXCONNECTS <https://curl.haxx.se/libcurl/c/CURLMOPT_MAXCONNECTS.html>`_.
-    The default is -1.
-
-    ``max_total_connections`` is the maximum number of active connections.
-    It affects libcurl  `CURLMOPT_MAX_TOTAL_CONNECTIONS <https://curl.haxx.se/libcurl/c/CURLMOPT_MAX_TOTAL_CONNECTIONS.html>`_.
-    It is ignored if the curl version is less than 7.30.
-    The default is ``0``, which allows libcurl to scale accordingly to easily handles count.
-    
-    The default option values are usually good enough but in rare cases it
-    might be good to set them. In that case here are two tips.
-
-    1. You may want to control the maximum number of sockets that a particular HTTP client uses simultaneously.
-       If a system passes many requests to distinct hosts, then libcurl cannot reuse sockets.
-       In this case setting ``max_total_connections`` may be useful,
-       since it causes curl to avoid creating too many sockets which would not be used anyway.
-
-    2. Do not set ``max_connections`` less than ``max_total_connections``
-       unless you are confident about your actions.
-       When ``max_connections`` is less then ``max_total_connections``, in some cases
-       libcurl will not reuse sockets for requests that are going to the same host.
-       If the limit is reached and a new request occurs, then 
-       libcurl will first create a new socket, send the request, wait for the first connection
-       to be free, and close it, in order to avoid exceeding the ``max_connections`` cache size.
-       In the worst case, libcurl will create a new socket for every request,
-       even if all requests are going to the same host.
-       See `this Tarantool issue on Github <https://github.com/tarantool/tarantool/issues/3945>`_
-       for details.
-
-    :return: a new HTTP client instance
+    Create an HTTP client instance.
+
+    :param table options: configuration options of the client (see :ref:`client_options <client_options>`)
+
+    :return: a new HTTP client instance (see :ref:`client_object <client_object>`)
     :rtype:  userdata
 
-    **Example:**
+    **Example**
+
+    ..  literalinclude:: /code_snippets/test/http_client/get_test.lua
+        :language: lua
+        :lines: 1
+
+
+..  _client_options:
+
+client_options
+~~~~~~~~~~~~~~
+
+..  class:: client_options
+
+    Configuration options of the client.
+    These options can be passed to the :ref:`http.client.new() <http-new>` function.
+
+    ..  _client_options-max_connections:
+
+    .. data:: max_connections
+
+        Specifies the maximum number of entries in the cache.
+        This option affects libcurl `CURLMOPT_MAXCONNECTS <https://curl.haxx.se/libcurl/c/CURLMOPT_MAXCONNECTS.html>`_.
+        The default is ``-1``.
+
+        **Example**
+
+        ..  code-block:: lua
+
+            local http_client = require('http.client').new({max_connections = 5})
+
+        **Note**
```

Comment:
why not an actual rst note?
`.. note::`

Reply:
I think the actual note for such a long description may distract the reader here. I agree that it is better to use the feature designed for such a use case, but two huge notes look worse here, I tried )